currency_rate_update/models/res_currency_rate_provider.py

```diff
@@ -318,7 +323,13 @@ def _scheduled_update(self):
                     provider.next_run - provider._get_next_run_period()
                 )
                 date_to = provider.next_run
-                provider._update(date_from, date_to, newest_only=True)
+                if (date_to != fields.Date.today()) or (
```

Comment:
This is some voodoo. If provider is requested data that it does not have, it should return none. I'm not sure I follow why higher-level module should consider close date of provider. Some providers don't even have one and have rate for every hour

Reply:
This is related to https://github.com/OCA/currency/pull/85#issuecomment-697223174